src/index.js

```diff
@@ -270,7 +271,15 @@ class UsersRouting extends Component {
         <Route path={`${base}/:id/patronblocks/edit/:patronblockid`} component={Routes.PatronBlockContainer} />
         <Route path={`${base}/:id/patronblocks/create`} component={Routes.PatronBlockContainer} />
         <Route path={`${base}/create`} component={Routes.UserEditContainer} />
-        <Route path={`${base}/custom-fields/edit`} exact component={Routes.EditCustomFields} />
+        <Route
+          path={`${base}/custom-fields/edit`}
+          exact
+          render={(props) => (
+            <IfPermission perm="ui-users.settings.customfields.edit">
```

Comment:
We shouldn't wrap the route component in `IfPermission`. Instead, please implement the logic of redirection to the root page of users app if the user doesn't have `customfields.edit` permission. This should be implemented by `Routes.EditCustomFields`.